### PR TITLE
[LMS-341] Namespace-based auth for Assignment/AssigmentSubmission

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

File: extension/EdFi.Ods.Extensions.LMSX/Artifacts/MsSql/Data/Security/0002-LMSX-AssignmentClaims.sql [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    )
+VALUES (
+    'assignment'
+    , 'assignment'
+    , 'http://ed-fi.org/ods/identity/claims/lmsx/assignment'
+    , @relationshipBasedDataResourceClaimId
+    , @applicationId
+    )
+
+-- Is assignment submission even necessary? We aren't putting a Namespace field on it. Should we?
```
Comment (stephenfuqua) [inline extension/EdFi.Ods.Extensions.LMSX/Artifacts/MsSql/Data/Security/0002-LMSX-AssignmentClaims.sql:33]: What's the implication?

A client app at Google pushes a new Assignment with the Google namespace, gets an ID back.

The client app now pushes a new Submission with that assignment, with no namespace, get an ID back (ID  = `84`).

Meanwhile Schoology does the same (ID = `85`), but makes a mistake in their code. They want to confirm they uploaded successfully and send `GET /lmsx/assignmentSubmission/84`  with the wrong Id. Are they able to retrieve that record?

Comment (bradbanister) [reply extension/EdFi.Ods.Extensions.LMSX/Artifacts/MsSql/Data/Security/0002-LMSX-AssignmentClaims.sql:33]: Looks like we should. I wasn't sure how strictly to be following the story, which didn't mention updating AssignmentSubmission.

Comment (bradbanister) [reply extension/EdFi.Ods.Extensions.LMSX/Artifacts/MsSql/Data/Security/0002-LMSX-AssignmentClaims.sql:33]: The thing though is that the AssignmentSubmission table will get a Namespace column via the foreign key reference to Assignment. I guess I'll explicitly declare it on AssignmentSubmission and merge.  ¯\_(ツ)_/¯

Comment (stephenfuqua) [reply extension/EdFi.Ods.Extensions.LMSX/Artifacts/MsSql/Data/Security/0002-LMSX-AssignmentClaims.sql:33]: FK reference alone seems reasonable to me.

**Comment by bradbanister** [comment]
Setup for QA:

With 5.2 ODS - symlink to the EdFi.Ods.Extensions.LMSX.New as EdFi.Ods.Extensions.LMSX in Ed-Fi-ODS-Implementation\Application using cmd with admin, something like "mklink /d EdFi.Ods.Extensions.LMSX C:\work\lms-toolkit\LMS-Toolkit\extension\EdFi.Ods.Extensions.LMSX.New"

initdev

Open Visual Studio. Ensure the EdFi.Ods.Extensions.LMSX project is added under Ed-Fi-Extensions. Do it via the symlink. Add the project as a reference to EdFi.Ods.WebApi

Build from VS. Note this will dump a bunch of generated directories in the repo, so you'll need to clean up when testing is finished.

Start the ODS/API from Visual Studio. It will populate the EdFi_Admin DB with sandbox data at start up.  Stop it after it's fully running.

EdFi_Admin DB prep:

dbo.Vendors should have "Test Admin" as VendorId 1
Add entry to dbo.VendorNamespacePrefixes with Vendor_VendorId 1 and NamespacePrefix for an LMS, like "uri://schoology.com"

Sandbox DB prep:

Start ODS/API from Visual Studio again.  When all is ready, go to the opened Swagger API page, select Resources then Authorize. Note the client id before clicking Authorize on the popup.

Find DB named EdFi_Ods_Sandbox_<client id>.  For ref integrity, add an entry of id 1 to both lmsx.LMSSourceSystemDescriptor and lmsx.AssignmentCategoryDescriptorId tables.

Here's an example for manual inserts into the Assignment table:

INSERT INTO [lmsx].[Assignment]
           ([AssignmentIdentifier]
           ,[Namespace]
           ,[LMSSourceSystemDescriptorId]
           ,[Title]
           ,[AssignmentCategoryDescriptorId]
           ,[AssignmentDescription]
           ,[StartDateTime]
           ,[EndDateTime]
           ,[DueDateTime]
           ,[MaxPoints]
           ,[SectionIdentifier]
           ,[LocalCourseCode]
           ,[SessionName]
           ,[SchoolYear]
           ,[SchoolId]
           ,[Discriminator]
           ,[CreateDate]
           ,[LastModifiedDate]
           ,[Id]
           ,[ChangeVersion])
     VALUES
           ('AssignmentIdentifier5'
           ,'uri://schoology.com'
           ,1
           ,'AssignmentTitle'
           ,1
           ,NULL
           ,NULL
           ,NULL
           ,NULL
           ,NULL
           ,'SectionIdentifier1'
           ,NULL
           ,NULL
           ,NULL
           ,NULL
           ,NULL
           ,CURRENT_TIMESTAMP
           ,CURRENT_TIMESTAMP
           ,'6097412a-cd24-4a01-805b-1ffba371a433'
           ,1)

Insert unique assignments with various Namespaces.  In the Swagger API page, run a get all and confirm results are only for namespaces with prefixes matching the NamespacePrefix from EdFi_Admin.dbo.VendorNamespacePrefixes

Try a few different assignment actions in the Swagger API.

Insert some assignment submissions and do the same. In your test data, make sure the namespaces match between a submission and its parent assignment.

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**